xml/src/main/java/org/ehcache/xml/ResourceConfigurationParser.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.xml;
+
+import org.ehcache.config.ResourcePool;
+import org.ehcache.config.ResourceUnit;
+import org.ehcache.config.SizedResourcePool;
+import org.ehcache.config.units.EntryUnit;
+import org.ehcache.config.units.MemoryUnit;
+import org.ehcache.core.config.SizedResourcePoolImpl;
+import org.ehcache.core.internal.util.ClassLoading;
+import org.ehcache.xml.exceptions.XmlConfigurationException;
+import org.ehcache.xml.model.Disk;
+import org.ehcache.xml.model.Heap;
+import org.ehcache.xml.model.MemoryType;
+import org.ehcache.xml.model.Offheap;
+import org.ehcache.xml.model.PersistableMemoryType;
+import org.ehcache.xml.model.ResourceType;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.SAXException;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.parsers.ParserConfigurationException;
+
+import static org.ehcache.xml.ConfigurationParser.CORE_SCHEMA_NAMESPACE;
+
+public class ResourceConfigurationParser {
+
+  private final Map<org.ehcache.config.ResourceType<?>, CacheResourceConfigurationParser> extensionResourceParsers = new HashMap<>();
```

Comment:
Why is this code pulled out here?  Whats so special about it?

Reply:
Those two maps you meant? One map is used by the parsing logic and the other one is for unparse.

Reply:
I mean the service and service creation configurations have their maps split between the translator and parser.  For resources you decided to have them both here, and use this class in both translator and parser.  Why the difference?